projects/ngx-charts-on-fhir/src/lib/analysis-cards/statistics/statistics.component.ts

```diff
@@ -22,22 +24,42 @@ export class StatisticsComponent extends AnalysisCardContent implements OnChange
     }
     return 5;
   }
-
+  maxDate: Date | string;
+  minDate: Date | string;
+  previousDate: Date | string;
+  monthsCount?: number;
   statistics: NameValuePair[] = [];
+  layers?: DataLayer[];
+  previousDataDates?: any[] = [];
+  constructor(private layerManager: DataLayerManagerService) {
+    super();
+    this.maxDate = new Date();
+    this.minDate = new Date();
+    this.previousDate = new Date();
+  }
 
   ngOnChanges(): void {
+    this.getMaxMinDate(this.visibleData)
+    this.layerManager.selectedLayers$.subscribe((layers) => {
+      this.layers = layers;
+      this.getPreviousDataFromLayers(this.layers, this.monthsCount || 0)
+    })
```

Comment:
Why does this component need to subscribe to all selected layers? This is supposed to be a "dumb" component that just performs calculations on the data that is passed in through its inputs. It should already have all of the information that it needs and it should not care which other layers are selected.